cupy/cuda/compiler.py

```diff
@@ -44,58 +42,30 @@ def __exit__(self, exc_type, exc_value, traceback):
         os.rmdir(self.path)
 
 
-def _run_nvcc(cmd, cwd):
-    try:
-        return subprocess.check_output(cmd, cwd=cwd, stderr=subprocess.STDOUT)
-    except subprocess.CalledProcessError as e:
-        msg = ('`nvcc` command returns non-zero exit status. \n'
-               'command: {0}\n'
-               'return-code: {1}\n'
-               'stdout/stderr: \n'
-               '{2}'.format(e.cmd, e.returncode, e.output))
-        raise RuntimeError(msg)
-    except OSError as e:
-        msg = 'Failed to run `nvcc` command. ' \
-              'Check PATH environment variable: ' \
-              + str(e)
-        raise OSError(msg)
-
-
-def nvcc(source, options=(), arch=None):
+def nvrtc(source, options=(), arch=None):
     if not arch:
         arch = _get_arch()
-    cmd = ['nvcc', '--cubin', '-arch', arch] + list(options)
+
+    options += ('-arch{}'.format(arch),)
 
     with TemporaryDirectory() as root_dir:
         path = os.path.join(root_dir, 'kern')
         cu_path = '%s.cu' % path
-        cubin_path = '%s.cubin' % path
 
         with open(cu_path, 'w') as cu_file:
             cu_file.write(source)
 
-        cmd.append(cu_path)
-        _run_nvcc(cmd, root_dir)
-
-        with open(cubin_path, 'rb') as bin_file:
-            return bin_file.read()
+        prog = Program(six.b(source), six.b(os.path.basename(cu_path)))
```

Comment:
I'm not familiar with bytes handling, but using six to convert it to bytes looks safe for both python2 and 3. Can `encode('latin-1')` be enough?